File: LeetCode/0121. Best Time to Buy and Sell Stock/solution.js
/**
 * 200 / 200 test cases passed
 * Status: Accepted
 * Runtime: 64 ms
 * 
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    [min, profit] = [Number.MAX_SAFE_INTEGER, 0]
    for (let p of prices) {
        min = Math.min(min, p)
        profit = Math.max(profit, p - min)
    }
    return profit
};

File: LeetCode/0121. Best Time to Buy and Sell Stock/solution.go
/**
 * 200 / 200 test cases passed
 * Status: Accepted
 * Runtime: 4 ms
 */
func maxProfit(prices []int) int {
    min, profit := int(^uint(0) >> 1), 0
    for _, p := range prices {
        if min > p {
            min = p
        } else {
            diff := p - min
            if diff > profit {
                profit = diff
            }
        }
    }
    return profit
}